scrape/scrape.go

```diff
@@ -303,6 +303,16 @@ func (sp *scrapePool) restartLoops(reuseCache bool) {
 		mrc                      = sp.config.MetricRelabelConfigs
 	)
 
+	var validationScheme model.ValidationScheme
+	switch sp.config.MetricNameValidationScheme {
+	case config.LegacyValidationConfig:
+		validationScheme = model.LegacyValidation
+	case config.UTF8ValidationConfig:
+		validationScheme = model.UTF8Validation
+	default:
+		validationScheme = model.LegacyValidation
+	}
```

Comment:
No need to check for `config.LegacyValidationConfig`. This can be simplified.

```suggestion
	validationScheme := model.LegacyValidation
	if sp.config.MetricNameValidationScheme == config.UTF8ValidationConfig {
		validationScheme = model.UTF8Validation
	}
```